### PR TITLE
nslookup.md

### LINKED ISSUE #309: create reference for nslookup

### PR DESCRIPTION
**Pull request by karankwatra**
# Advanced HW 5 submission

Closes #309

Name: Karan Kwatra
Uniqname: kwatraka

### CODE REVIEW

File: _commands/tools/nslookup.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+---
+
+nslookup
+-------
+
+nslookup which stands for "name server lookup" is used to query internet name servers interactively for information. 
+
+~~~ bash
+nslookup [-option ...] [host-to-find | -[server]]
```
Comment (mterwill) [inline _commands/tools/nslookup.md]: This is a little too `man`page-y for me. How about something as simple as

```bash
$ nslookup www.google.com
```

File: _commands/tools/nslookup.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+Name:	microsoft.com
+Address: 23.100.122.175
+Name:	microsoft.com
+Address: 23.96.52.53
+Name:	microsoft.com
+Address: 191.239.213.197
+Name:	microsoft.com
+Address: 104.40.211.35
+Name:	microsoft.com
+Address: 104.43.195.251
```
Comment (mterwill) [inline _commands/tools/nslookup.md]: This doesn't look quite right. Did you preview it in your development environment? I would put this stdout in the code block as well.

image

File: _commands/tools/nslookup.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+*Here 141.211.125.17 is the address of our system's DNS. #53 indicates that we are communicating with it on port 53, 
+which is the standard port number DNS's use to accept queries. 
+*Below this is the lookup info of microsoft.com. Here we get 5 different addresses which indicates that when you access
```
Comment (mterwill) [inline _commands/tools/nslookup.md]: These aren't showing up as bullet points. I think you need to put a space between the asterisk and content.

File: _commands/tools/nslookup.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+a.ns.facebook.com	internet address = 69.171.239.12
+a.ns.facebook.com	has AAAA address 2a03:2880:fffe:c:face:b00c:0:35
+b.ns.facebook.com	internet address = 69.171.255.12
+b.ns.facebook.com	has AAAA address 2a03:2880:ffff:c:face:b00c:0:35
+ns3.facebook.com	internet address = 66.220.151.20
+ns3.facebook.com	has AAAA address 2a03:2880:fffe:c:face:b00c:0:35
+ns4.facebook.com	internet address = 69.171.245.32
+ns4.facebook.com	has AAAA address 2a03:2880:ffff:c:face:b00c:0:35
+ns5.facebook.com	internet address = 66.220.145.65
+ns5.facebook.com	has AAAA address 2a03:2880:fffe:c:face:b00c:0:35
```
Comment (mterwill) [inline _commands/tools/nslookup.md]: Same thing here; this also isn't previewing correctly.

Also: touch on `AAAA` records and IPv6?

**Comment by mterwill** [comment]
@karankwatra: please follow up by this Friday or we'll close the PR and won't be able to give you credit for the assignment.

**Comment by karankwatra** [comment]
Let me know if you would like any more changes to be done.

File: _commands/tools/nslookup.md [hunk 5]
```diff
@@ -2,24 +2,23 @@
 ---
 
 nslookup
--------
-
-nslookup which stands for "name server lookup" is used to query internet name servers interactively for information.
```
Comment (mterwill) [inline _commands/tools/nslookup.md]: Why'd you get rid of the explanation here? I think it's valuable. At least say something about how you can use `nslookup` to translate DNS names to IP addresses and vice versa.

File: _commands/tools/nslookup.md [hunk 6]
```diff
@@ -2,7 +2,9 @@
 ---
 
 nslookup
-------- 
+-------
+-------
+nslookup which stands for "name server lookup" is used to query internet name servers interactive
```
Comment (mterwill) [inline _commands/tools/nslookup.md]: This isn't grammatically correct. While you're fixing, I'd prefer if you'd include something more accessible to the course as I suggested in my previous review... Most people probably don't know what a name server is.

**Comment by mterwill** [comment]
Thanks for working through this. LGTM.

**Pull request merged**